spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/core/subscriber/PubSubSubscriberTemplate.java

```diff
@@ -190,13 +191,13 @@ public Subscriber subscribe(String subscription,
 	 * Pulls messages synchronously, on demand, using the pull request in argument.
 	 *
 	 * @param pullRequest pull request containing the subscription name
+	 * @param subscriberStub subscriber stub
 	 * @return the list of {@link AcknowledgeablePubsubMessage} containing the ack ID, subscription
 	 * and acknowledger
 	 */
-	private List<AcknowledgeablePubsubMessage> pull(PullRequest pullRequest) {
+	private List<AcknowledgeablePubsubMessage> pull(PullRequest pullRequest, SubscriberStub subscriberStub) {
```

Comment:
Why do you need to pass `subscriberStub` into the method? Doesn't `pullRequest` contain the subscription name which you can use to lookup the stub in the map?

Reply:
Great catch. Sorry, passing in `subscriberStub` is unnecessary here.